astropy/table/table.py

```diff
@@ -1312,9 +1313,16 @@ def __setitem__(self, item, value):
     def __delitem__(self, item):
         if isinstance(item, six.string_types):
             self.remove_column(item)
-        elif isinstance(item, tuple):
+
+        elif _is_int(item):
+            self.remove_row(item)
+
+        elif isinstance(item, (list, tuple, np.ndarray)) and all(isinstance(x, six.string_types) for x in item):
             self.remove_columns(item)
 
+        elif isinstance(item, (list, np.ndarray)) and all(_is_int(x) for x in item):
```

Comment:
For performance do:
```
elif isinstance(item, (list, np.ndarray)) and np.asarray(x).dtype.kind == 'i':
```
This does create a temporary intermediate array, but in this case I think that is better than dropping to a slow pure Python loop.  For a million-long `item` this method is about 8 times faster.